git.md

```diff
@@ -195,6 +195,26 @@ Fix nasty little bug ISSUE-51 #resolve #time 2h30m
 * Keep the history *clean* and *simple*. Before pushing any branch to the
   public always ensure that it conforms to the style-guide. If it doesn't:
   squash, rebase, reword commits as necessary.
+  
+* When merging multiple small commits into a single change, make sure to
+  keep high locality of the changes. In another words, if you're changing,
+  let's say, some elements of the UI, group changes
+  that occur in small vicinity of each other.
+
+* When merging multiple changes into one, make sure to _do not exceed
+  limit of 10 changes in one commit_. Additionally, always include
+  ticket/issue numbers in the commit log. For example:
+
+  ```git
+  Improve user-experience in subscription views PROJ-10
```

Comment:
Personally I'm not in favour of creating commits that relate to several JIRA issues. For me it's fogging the boundaries dividing changes that are related to specific tasks/bugs etc.

Reply:
See me response to @dragonee above.

Reply:
I see the point, but I would emphasize that this is not recommended and should be done as a last resort: otherwise people will abuse this rule.

Reply:
On the other hand - sometimes tickets are created in a very granular way (e.g. separate ticket for each letter of CRUD), and so there no logical way to separate them, and so a single commit is related to at least four tickets, in a way that doesn't seem last resort-ish to me

Reply:
"separate ticket for each letter of CRUD" - This sounds to me like 4 pull requests for 4 features.

Reply:
To be completely clear, this proposition is about really small commits.